source/host_resolver.c

```diff
@@ -547,6 +567,61 @@ static inline void process_records(
     }
 }
 
+static void s_purge_host_cache_callback_task(struct aws_task *task, void *arg, enum aws_task_status status) {
+    (void)status;
+    struct host_purge_callback_options *options = arg;
+    aws_mem_release(options->allocator, task);
+    options->on_host_purge_complete(options->on_host_purge_complete_user_data);
+    aws_mem_release(options->allocator, options);
+}
+
+static int resolver_purge_host_cache(const struct aws_host_resolver_purge_host_options *options) {
```

Comment:
This function doesn't fail at all, can we use void return type?

Rename to `s_resolver_purge_host_cache` as it's a static function. If there are other functions in this code based not following that naming pattern, please fix them as well.

Reply:
I thought about using a void, however, this is an interface. In future, there might be other implementations that throw an error. I have fixed the naming.